src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -559,7 +573,8 @@ private void updateAccountAndRoles(Study study, Account account, StudyParticipan
         account.setNotifyByEmail(participant.isNotifyByEmail());
         account.setDataGroups(participant.getDataGroups());
         account.setLanguages(participant.getLanguages());
-        account.setMigrationVersion(AccountDao.MIGRATION_VERSION);
+        account.setSynapseUserId(participant.getSynapseUserId());
```

Comment:
Should this be in updateIdentifiers() instead?

Reply:
It didn't seem like we needed to prevent updates to this value after account creation, because it isn't validated (if it's wrong, you just can't sign in to that study with your Synapse account until it's fixed).

Reply:
The reason why that exists is to force users to pass in their credentials in order to change their email or phone number, to prevent someone from stealing your session, hijacking your account, and then using that to take over your account permanently.

I can see this not being an issue with externalId, because externalId is an arbitrary string, and there is no external ID equivalent of reset password email or email sign-in.

With Synapse User ID, it’s a little trickier, since if the user doesn’t have a password, you can’t ask for a password to change the Synapse User ID it’s associated with. That said, I still think the risk of someone hijacking the Bridge account is high enough that we should prevent this from happening. It would be safer to block updating Synapse User ID altogether, except through an admin API. I’ll check with Aaron.

Reply:
This is moved to the updateIdentifiers API.